packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -1,11 +1,36 @@
 const contentful = require(`contentful`)
 const _ = require(`lodash`)
-
+const fs = require(`fs-extra`)
+const path = require(`path`)
 const normalize = require(`./normalize`)
 
-module.exports = async ({ spaceId, accessToken, host, syncToken, environment }) => {
+const cacheFilename = `contentful-result.json`
+
+async function writeCacheFile(dir, result) {
+  try {
+    await fs.writeJson(path.join(dir, cacheFilename), result)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+module.exports = async ({ spaceId, accessToken, host, syncToken, cacheDir, environment }) => {
   // Fetch articles.
   console.time(`Fetch Contentful data`)
+
+  // If the user knows they are offline, serve them cached result
+  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true`) {
+    console.log(`Using Contentful Offline cache ⚠️`)
+    console.log(`Cache may be invalidated if you edit package.json, gatsby-node.js or gatsby-config.js files`)
```

Comment:
Maybe those 2 lines could be moved after we read cached data, so it would would print out only if we actually can use it? Not totally sure on this, but this might be better develop experience?